File: backend-Java/src/main/java/pl/kacste/patient/PatientService.java
package pl.kacste.patient;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import javax.persistence.EntityNotFoundException;

@Service
@RequiredArgsConstructor
class PatientService {

    private final PatientRepository patientRepository;

    public String createPatient(Patient patient){
        return patientRepository.save(patient).getIdentityNumber();
    }

    public Patient findOne(String identityNumber){
        return patientRepository
                .findById(identityNumber)
                .orElseThrow(() -> new EntityNotFoundException("Could not found patient with idenity number " + identityNumber));
    }
}

File: backend-C#/DoctorsOffice/Patients/PatientService.cs

namespace DoctorsOffice.Patients
{
    public class PatientService
    {
        private readonly PatientRepository _patientRepository;

        public PatientService(PatientRepository patientRepository)
        {
            _patientRepository = patientRepository;
        }

        public string CreatePatient(Patient patient)
        {
            return _patientRepository.Save(patient).IdentityNumber;
        }

        public Patient FindOne(string identityNumber)
        {
            return _patientRepository.FindById(identityNumber);
        }
    }
}